src/monotonic/count_rects.rs

```diff
@@ -0,0 +1,65 @@
+//! # Count Rectangles
+
+use crate::monotonic::mono_range::mono_range;
+use crate::monotonic::mono_st::mono_st;
+
+/// grid is a n-by-m boolean array
+///
+/// Gets an (n+1)-by-(m+1) vec cnt where cnt\[i\]\[j\] = the number of times an i-by-j subrectangle
+/// appears in the grid such that all i*j cells in the subrectangle are true.
+///
+/// cnt\[i\]\[0\] and cnt\[0\]\[j\] will contain garbage numbers.
+///
+/// # Example
+/// ```
+/// use programming_team_code_rust::monotonic::count_rects::count_rects;
+///
+/// let mut cnt = count_rects(&[vec![true, true, false],
+///                             vec![true, true, true]]);
+///
+/// // remove garbage values
+/// cnt.drain(0..1);
+/// for i in 0..cnt.len() {
+///    cnt[i].drain(0..1);
+/// }
+///
+/// assert_eq!(cnt, [[5, 3, 1],
+///                  [2, 1, 0]]);
+///
+/// assert!(std::panic::catch_unwind(|| count_rects(&[])).is_err());
+/// ```
+///
+/// # Complexity
+/// - Time: O(n * m)
+/// - Space: O(n * m)
+pub fn count_rects(grid: &[Vec<bool>]) -> Vec<Vec<i64>> {
```

Comment:
about the return type:

it's signed because of how the prefix sums work (there's intermediate negative values), although we could have it be unsigned, and use wrapping_add/wrapping_sub if we wanted

I was thinking about having it return `Vec<Vec<i32>>`; but I chose 64-bits because then naturally to avoid casts, anything you calculate based on this, you'll avoid casts, and use 64 bit ints, and then avoid overflow

Reply:
I would vote to use `isize` I think

Reply:
but with `isize` : to know whether overflow might happen, you have to look up whether the judge uses 32 bit or 64 bit machines

Reply:
correct